### PR TITLE
Automatin (I)Engine - depcreate and add methods dealing with alarms

### PR DESCRIPTION
**Pull request by RobbeDGH**
[[RN40773](https://intranet.skyline.be/DataMiner/Lists/Release%20Notes/DispForm2.aspx?ID=40773)] [[DCP246488](https://collaboration.skyline.be/task/246488)]

### CODE REVIEW

File: src/DataMiner/SLNetTypes/Info/IDObjects/AlarmTreeID.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+using Skyline.DataMiner.Net.Messages.SLDataGateway;
+using System;
+
+namespace Skyline.DataMiner.Net
+{
+    /// <summary>
+    /// Represents a reference to one specific alarm tree.
+    /// </summary>
+    [Serializable]
+    public sealed class AlarmTreeID //: DMAObjectRef, IEquatable<AlarmTreeID>
```
Comment (wdm-slc) [inline src/DataMiner/SLNetTypes/Info/IDObjects/AlarmTreeID.cs:11]: do we also need to list the properties like ElementID and RootAlarmID ?

**Comment by RobbeDGH** [comment]
@wdm-slc, to see the resulting docs from this you need to use buildDocs.cmd instead of the raw docfx build command

File: src/DataMiner/SLNetTypes/Info/IDObjects/AlarmTreeID.cs (hunk 1 above)
Comment (RobbeDGH) [reply src/DataMiner/SLNetTypes/Info/IDObjects/AlarmTreeID.cs:11]: seems like a good idea yeah

**Comment by RobbeDGH** [comment]
Note there is somthing wrong with displaying the AlarmID class, not sure what.
This is not fixed.

Think it might be related to the namespace Skyline.DataMiner.Net.Messages.SLDataGateway being present in multiple dlls

**Review by wdm-slc — APPROVED** [review]
lgtm

File: src/DataMiner/SLNetTypes/Info/IDObjects/AlarmTreeID.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    {
+        // default ctor is required for protobuf
```
Comment (MariekeGO) [inline src/DataMiner/SLNetTypes/Info/IDObjects/AlarmTreeID.cs:13]: Is this correct, or is "ctor" a typo?

Comment (RobbeDGH) [reply src/DataMiner/SLNetTypes/Info/IDObjects/AlarmTreeID.cs:13]: It is correct. 'ctor' is often used as a shorthand for 'constructor'.

I copied these parts over from the official source.
Perhaps we should use the full word for public documentation though. 
Up to you

Comment (MariekeGO) [reply src/DataMiner/SLNetTypes/Info/IDObjects/AlarmTreeID.cs:13]: indeed, I'll adapt this to constructor so it's clear for everyone

**Review by MariekeGO — APPROVED** [review]

**Pull request merged**